File: daily_python/2348_Number_of_Zero_Filled_Subarrays.py
import os
import sys
from typing import List

class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        ans = 0
        cnt = 0
        for num in nums:
            cnt = 0 if num != 0 else cnt + 1
            ans += cnt
        return ans

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(nums)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums in dataset:
        results.append(solution.zeroFilledSubarray(nums))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2025/2348_Number_of_Zero_Filled_Subarrays.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        long long cnt = 0;
        int contiguous = 0;
        for (int num : nums){
            if (num == 0) contiguous++;
            else contiguous = 0;
            cnt += contiguous;
        }
        return cnt;
    }
};

int main(){
    vector<int> nums = {1,3,0,0,2,0,0,4};
    // vector<int> nums = {0,0,0,2,0,0};
    // vector<int> nums = {2,10,2019};
    Solution S;
    cout << S.zeroFilledSubarray(nums) << endl;
    return 0;
}